scipy/stats/tests/test_mstats_basic.py

```diff
@@ -869,6 +869,13 @@ def test_regress_simple():
     assert_almost_equal(result.stderr, 0.002395781449783862)
     assert_almost_equal(result.intercept_stderr, 0.13866936078570702)
 
+
+def test_linregress_identical_x():
+    x = np.zeros(10)
+    y = np.random.rand(10)
+    assert_raises(ValueError, mstats.linregress, x, y)
```

Comment:
Use the match statement to be sure to catch the correct exception.
```suggestion
    msg = "Cannot calculate a linear regression"
    with assert_raises(ValueError, match=msg):
        mstats.linregress(x, y)
```

Reply:
Good idea, I've changed it now